src/box/box.cc

```diff
@@ -796,14 +797,25 @@ box_check_replication(void)
 static int
 box_check_listen(void)
 {
-	int count = cfg_getarr_size("listen");
-	for (int i = 0; i < count; i++) {
-		const char *source = cfg_getarr_elem("listen", i);
-		assert(source != NULL);
-		if (box_check_uri(source, "listen") != 0)
-			return -1;
+	struct uri_set uri_set;
+	cfg_get("listen");
+	int rc = luaL_uri_set_create(tarantool_L, &uri_set);
+	lua_pop(tarantool_L, 1);
+	if (rc != 0) {
+		diag_set(ClientError, ER_CFG, "listen",
+			 "invalid input format");
```

Comment:
Bad error message - it's impossible to figure what happened here.

Reply:
You think we need to set diag in `luaT_uri_set_create` or it is enough to print common message for all cases?

Reply:
Diag would probably be the best, but I'm not sure. You may also ask what @alyapunov thinks - do we need verbose errors or is it enough to just say that the uri is invalid?

Reply:
fixed